addons/website_sale_stock/models/product.py

```diff
@@ -0,0 +1,17 @@
+# -*- coding: utf-8 -*-
+from odoo import api, models, _
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    @api.multi
+    def _product_availabity(self):
+        self.ensure_one()
+        is_available = True
+        if self.type=='product':
+            is_available = False
+            for route in self.sudo().mapped('route_ids.name'):
+                if route in [_("Manufacture"), _("Buy")]:
```

Comment:
On the chatter I said that the condition is Make-to-Order.
Having the "Buy" or "Manufacture" routes doesn't mean that you are working in just-in-time delivery.